services/postgres/alembic/env.py

```diff
@@ -67,12 +67,12 @@ def apply_hasura_metadata(context):
 
     fn = context.get_context().opts.get("fn")
     if fn and fn.__name__ == "upgrade":
-        if context.get_context().opts.get("destination_rev") == "heads":
+        if script.revision == context.script.get_current_head():
             hasura_revision = None
         else:
-            hasura_revision = script.down_revision
+            hasura_revision = script.up_revision
     elif fn and fn.__name__ == "downgrade":
-        hasura_revision = script.revision
+        hasura_revision = script.down_revision
```

Comment:
why are you having to touch this file?

Reply:
There's a small bug in this code (two actually) which is exposed by destructive migrations like this one -- essentially that it only runs properly if you pass the string `"heads"` as the alembic target. Our unit tests for migrations don't pass `"heads"`; they pass an explicit number of migrations to run (`"+2"`). Those didn't trip this logic correctly and attempted to apply a hasura metadata file that was not compatible with the database state.